### DISCUSSION TITLE
Failed when trying to install deepmd-kit2.1.3&CUDA11.6 with conda

### DISCUSSION THREAD

**Discussion by gdbhcxhmjk-z** [deepmodeling/deepmd-kit#1810] [Q&A]
Find environment conflicts when trying to intall deepmd-kit v2.1.3 &cuda11.6.
The installation command is : conda create -n deepmd deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps cudatoolkit=11.6 horovod -c https://conda.deepmodeling.com

The conflicts information is shown as follows:
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: -
Found conflicts! Looking for incompatible packages.                                                                                                                                                                                  failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package mpich conflicts for:
lammps -> mpich[version='>=3.3.2,<3.4.0a0|>=3.3.2,<4.0a0']
horovod -> mpi4py -> mpich[version='>=3.3,<3.4.0a0']
libdeepmd==2.1.3[build=*gpu] -> mpich[version='>=3.3.2,<3.4.0a0|>=3.3.2,<4.0a0']
lammps -> pylammps==20220623.0=pyh29e1a88_0 -> mpich
horovod -> mpich[version='>=3.3.2,<3.4.0a0|>=3.3.2,<4.0a0']

Package numpy conflicts for:
deepmd-kit==2.1.3[build=*gpu] -> numpy
deepmd-kit==2.1.3[build=*gpu] -> h5py -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|>=1.11.3,<2.0a0|>=1.14.6,<2.0a0|>=1.16.6,<2.0a0|>=1.21.2,<2.0a0|1.9.*|1.8.*|1.7.*|1.6.*|1.5.*|>=1.15.1,<2.0a0|>=1.9.3,<2.0a0|>=1.21.5,<2.0a0']

Package libstdcxx-ng conflicts for:
libdeepmd==2.1.3[build=*gpu] -> libstdcxx-ng[version='>=7.5.0']
libdeepmd==2.1.3[build=*gpu] -> cudnn -> libstdcxx-ng[version='>=7.3.0']

Package sqlite conflicts for:
horovod -> python[version='>=3.9,<3.10.0a0'] -> sqlite[version='>=3.24.0,<4.0a0|>=3.25.2,<4.0a0|>=3.25.3,<4.0a0|>=3.26.0,<4.0a0|>=3.27.2,<4.0a0|>=3.29.0,<4.0a0|>=3.30.0,<4.0a0|>=3.30.1,<4.0a0|>=3.31.1,<4.0a0|>=3.32.3,<4.0a0|>=3.33.0,<4.0a0|>=3.35.4,<4.0a0|>=3.36.0,<4.0a0|>=3.38.0,<4.0a0|>=3.38.2,<4.0a0|>=3.38.3,<4.0a0|>=3.37.2,<4.0a0']
deepmd-kit==2.1.3[build=*gpu] -> python[version='>=3.8,<3.9.0a0'] -> sqlite[version='>=3.30.0,<4.0a0|>=3.30.1,<4.0a0|>=3.31.1,<4.0a0|>=3.32.3,<4.0a0|>=3.33.0,<4.0a0|>=3.35.4,<4.0a0|>=3.36.0,<4.0a0|>=3.38.0,<4.0a0|>=3.38.3,<4.0a0|>=3.38.2,<4.0a0|>=3.37.2,<4.0a0']
libdeepmd==2.1.3[build=*gpu] -> libtensorflow_cc=2.9[build=cuda*] -> sqlite[version='>=3.38.3,<4.0a0']

Package h5py conflicts for:
deepmd-kit==2.1.3[build=*gpu] -> h5py
deepmd-kit==2.1.3[build=*gpu] -> tensorflow=2.9[build=cuda*] -> h5py[version='>=2.9.0']

Package wrapt conflicts for:
deepmd-kit==2.1.3[build=*gpu] -> tensorflow=2.9[build=cuda*] -> wrapt[version='>=1.11.0']
horovod -> tensorflow=2.9[build=cuda*] -> wrapt[version='>=1.11.0|>=1.11.1|>=1.11.2']

Package mpi4py conflicts for:
lammps -> pylammps==20220623.0=pyh29e1a88_0 -> mpi4py
horovod -> mpi4py

Package python conflicts for:
horovod -> cloudpickle -> python[version='2.6.*|2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3|>=3.5|>=3.6|>=3.6,<3.7.0a0|>=3.5,<3.6.0a0|3.4.*|3.3.*']
lammps -> pylammps==20220623.0=pyh29e1a88_0 -> python[version='>=3']
deepmd-kit==2.1.3[build=*gpu] -> python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']
horovod -> python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0']
deepmd-kit==2.1.3[build=*gpu] -> dargs[version='>=0.2.6'] -> python[version='2.6.*|2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.6|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|3.4.*|3.3.*|>=3.5']

Package _openmp_mutex conflicts for:
libdeepmd==2.1.3[build=*gpu] -> libgcc-ng[version='>=7.5.0'] -> _openmp_mutex[version='>=4.5']
cudatoolkit=11.6 -> libgcc-ng[version='>=7.5.0'] -> _openmp_mutex[version='>=4.5']
horovod -> libgcc-ng[version='>=7.5.0'] -> _openmp_mutex[version='>=4.5']
lammps -> libgcc-ng[version='>=7.5.0'] -> _openmp_mutex[version='>=4.5']
deepmd-kit==2.1.3[build=*gpu] -> libgcc-ng[version='>=7.5.0'] -> _openmp_mutex[version='>=4.5']

Package libopenblas conflicts for:
deepmd-kit==2.1.3[build=*gpu] -> numpy -> libopenblas[version='>=0.2.20,<0.2.21.0a0|>=0.3.2,<0.3.3.0a0|>=0.3.20,<1.0a0|>=0.3.3,<1.0a0']
lammps -> plumed[version='>=2'] -> libopenblas[version='>=0.3.10,<1.0a0|>=0.3.18,<1.0a0|>=0.3.13,<1.0a0']

Package libgcc-ng conflicts for:
libdeepmd==2.1.3[build=*gpu] -> libgcc-ng[version='>=7.5.0']
libdeepmd==2.1.3[build=*gpu] -> cudnn -> libgcc-ng[version='>=7.3.0']

Package cudatoolkit conflicts for:
cudatoolkit=11.6
deepmd-kit==2.1.3[build=*gpu] -> cudatoolkit[version='10.2|10.2.*|>=11.6,<12']
horovod -> tensorflow-base=2.5.0[build=gpu*] -> cudatoolkit[version='10.0|10.0.*|10.1.*|11.3.*|9.2|9.2.*']
horovod -> cudatoolkit[version='*|10.1|10.1.*|10.2|10.2.*|>=11.6,<12|>=11.3,<12|>=11.3.1,<11.4.0a0*|>=10.1.243,<10.2.0a0*']

Package jpeg conflicts for:
libdeepmd==2.1.3[build=*gpu] -> libtensorflow_cc=2.9[build=cuda*] -> jpeg[version='>=9e,<10a']
deepmd-kit==2.1.3[build=*gpu] -> tensorflow=2.9[build=cuda*] -> jpeg[version='>=9e,<10a']
lammps -> jpeg[version='>=9e,<10a']
horovod -> tensorflow=2.9[build=cuda*] -> jpeg[version='>=9b,<10a|>=9d,<10a|>=9e,<10a']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.17=0
  - feature:|@/linux-64::__glibc==2.17=0
  - cudatoolkit=11.6 -> __glibc[version='>=2.17,<3.0.a0']
  - cudatoolkit=11.6 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - deepmd-kit==2.1.3[build=*gpu] -> __glibc[version='>=2.17']
  - deepmd-kit==2.1.3[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - horovod -> __glibc[version='>=2.17']
  - horovod -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - lammps -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - libdeepmd==2.1.3[build=*gpu] -> __cuda

Your installed version is: 2.17

**Comment by njzjz** [comment]
You may not install the display driver. Read conda documentation [Managing virtual packages](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html).

**Reply by gdbhcxhmjk-z** [reply]
Thanks for your advice. I've checked conda virtual packages, and there is no virtual package "_cuda".  The information is shown as follows:
         python version : 3.8.5.final.0
       virtual packages : __linux=3.10.0=0
                          __glibc=2.17=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /data/apps/anaconda/2020.11  (read only)
How can I set the _cuda virtual package?

**Reply by njzjz** [reply]
Conda checks CUDA driver library `libcuda.so`. Or, read [Overriding detected packages](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#overriding-detected-packages).

**Comment by Zhang-pchao** [comment]
Please try using the offline installer (https://github.com/deepmodeling/deepmd-kit/releases/tag/v2.1.3)
